### PR TITLE
feat: support eureka registry

### PR DESCRIPTION
**Pull request by Charlie17Li**
### Ⅰ. Describe what this PR did
support eureka service registry

### Ⅱ. Does this pull request fix one issue?

fixes: #383

### CODE REVIEW

**Comment by johnlanni** [comment]
@Charlie17Li Please add copyright header to the new go files.

**Comment by johnlanni** [comment]
@Charlie17Li Please fix the e2e

**Comment by Charlie17Li** [comment]
> @Charlie17Li Please fix the e2e

ok

File: registry/eureka/watcher.go [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+	return result
+}
+
+func generateServiceEntry(app *fargo.Application) *v1alpha3.ServiceEntry {
+	portList := make([]*v1alpha3.Port, 0)
+	endpoints := make([]*v1alpha3.WorkloadEntry, 0)
+
+	for _, instance := range app.Instances {
+		protocol := common.HTTP
```
Comment (johnlanni) [inline registry/eureka/watcher.go:267]: We should support other protocols, such as gRPC, which you can parse from the service's metadata if present.

**Review by johnlanni — CHANGES_REQUESTED** [review]

File: registry/eureka/watcher.go (hunk 1 above)
Comment (Charlie17Li) [reply registry/eureka/watcher.go:267]: maybe we can get protocol type from `instance.Metadata`

```
if val, _ := instance.Metadata.GetString("protocol"); val != "" {
			protocol = common.ParseProtocol(val)
		}
```

Comment (johnlanni) [reply registry/eureka/watcher.go:267]: That's ok

Comment (johnlanni) [reply registry/eureka/watcher.go:267]: You should check the protocol is valid, if not, should use the http as default.

**Review by johnlanni — APPROVED** [review]
Great! LGTM

**Pull request merged**